### PR TITLE
New package: tree-sitter

### PR DESCRIPTION
**Pull request by wsw0108**

### CODE REVIEW

File: mingw-w64-tree-sitter/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+url="https://tree-sitter.github.io/"
+license=('MIT')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc")
+source=("${_realname}-${pkgver}.tar.gz::https://github.com/tree-sitter/tree-sitter/archive/refs/tags/v${pkgver}.tar.gz"
+        001-mingw-build.patch)
+sha256sums=('12a3f7206af3028dbe8a0de50d8ebd6d7010bf762db918acae76fc7585f1258d'
+            '19d2a1608ca5c6138c228b97815c48ac95be2bec1265c207fdc0950a65b8ebad')
+
```
Comment (LABPBS) [inline mingw-w64-tree-sitter/PKGBUILD:18]: noextract=("${_realname}-${pkgver}.tar.gz")

File: mingw-w64-tree-sitter/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+prepare() {
+  [[ -d ${_realname}-${pkgver} ]] && rm -rf ${_realname}-${pkgver}
+  MSYS=winsymlinks:lnk tar zxf "${srcdir}/${_realname}-${pkgver}.tar.gz"
```
Comment (LABPBS) [inline mingw-w64-tree-sitter/PKGBUILD:21]: tar zxf "${srcdir}/${_realname}-${pkgver}.tar.gz" || true

Comment (Biswa96) [reply mingw-w64-tree-sitter/PKGBUILD:21]: ZIP file can also be used as a workaround. For example, see mingw-w64-ffmpeg-python/PKGBUILD file.

**Comment by wsw0108** [comment]
This package directory should be change to libtree-sitter, because there has a cli tool in rust called tree-sitter

File: mingw-w64-tree-sitter/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            '19d2a1608ca5c6138c228b97815c48ac95be2bec1265c207fdc0950a65b8ebad')
+noextract=("${_realname}-${pkgver}.zip")
+
+prepare() {
+  cd "${srcdir}"
+  [[ -d ${_realname}-${pkgver} ]] && rm -rf ${_realname}-${pkgver}
+  unzip "${_realname}-${pkgver}.zip"
+  cd "${srcdir}/${_realname}-${pkgver}"
+  patch -p1 -i ${srcdir}/001-mingw-build.patch
+}
```
Comment (lazka) [inline mingw-w64-tree-sitter/PKGBUILD:26]:
```suggestion
cargo fetch --locked
}
```

File: mingw-w64-tree-sitter/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+build() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  make clean
+  make
+  cd "${srcdir}/${_realname}-${pkgver}"
+  cargo build --release
```
Comment (lazka) [inline mingw-w64-tree-sitter/PKGBUILD:33]:
```suggestion
  cargo build --locked --offline --release
```

**Pull request merged**